src/main/java/org/gridsuite/directory/server/repository/DirectoryElementRepository.java

```diff
@@ -22,34 +21,25 @@
 @Repository
 public interface DirectoryElementRepository extends JpaRepository<DirectoryElementEntity, UUID> {
 
-    @Query("SELECT d FROM DirectoryElementEntity d  WHERE d.parentId = :parentId AND (d.isPrivate='false' or d.owner=:userId)")
-    List<DirectoryElementEntity> findDirectoryContentByUserId(UUID parentId, String userId);
+    List<DirectoryElementEntity> findAllByParentId(UUID parentId);
 
     @Query("SELECT d FROM DirectoryElementEntity d  WHERE d.parentId IS NULL AND d.type = 'DIRECTORY' AND (d.isPrivate='false' or d.owner=:owner)")
     List<DirectoryElementEntity> findRootDirectoriesByUserId(String owner);
 
+    @Query("SELECT d FROM DirectoryElementEntity d  WHERE d.parentId IS NULL AND d.type = 'DIRECTORY' AND d.name=:name")
+    List<DirectoryElementEntity> findRootDirectoriesByName(String name);
```

Comment:
we can have multiple root with the same name ?

Reply:
Yes  it's planned
See discussions in Slack